src/Pages/TodoPage.tsx

```diff
@@ -0,0 +1,255 @@
+
+  const [isArrow, setIsArrow] = useState(false);
+
+  const isProcessing = useCallback((id: number) => {
+    return !!processingIds.includes(id) || id === 0;
+  }, [processingIds]);
+
+  const onShowArrow = useMemo(() => {
+    return setIsArrow(todos.every((todo: Todo) => todo.completed));
+  }, [todos]);
+
+  const visibleTodos = useMemo(() => {
+    return filteredTodos(todos, filter);
+  }, [todos, filter]);
+
+  const getTodosFromServer = async () => {
+    if (!user) {
+      return;
+    }
+
+    try {
+      const todosFromServer = await getTodos(user.id);
+
+      setTodos(todosFromServer);
+    } catch {
+      setCurrentError(Error.UPLOAD);
+    }
+  };
+
+  useEffect(() => {
+    if (!user) {
+      return;
+    }
+
+    setUserName(user.name);
+    getTodosFromServer();
+  }, [user]);
+
+  const onAdd = useCallback((todo: Todo) => {
+    setIsCreated(true);
+
+    const toAddTodo = async () => {
+      try {
+        const addedTodo = await addTodo(todo);
+
+        setTodos([...todos, addedTodo]);
+        dispatch({ type: ReducerTempTodoType.RESET });
+      } catch {
+        setCurrentError(Error.ADD);
+      } finally {
+        setIsCreated(false);
+      }
+    };
+
+    return toAddTodo();
+  }, [todos]);
+
+  const onDelete = useCallback((todoIdToDelete: number) => {
+    setProcessingIds(prev => [...prev, todoIdToDelete]);
+
+    const toRemoveTodo = async () => {
+      try {
+        await removeTodo(todoIdToDelete);
+
+        await getTodosFromServer();
+      } catch {
+        setCurrentError(Error.REMOVE);
+      } finally {
+        setProcessingIds(prev => prev.filter(id => id !== todoIdToDelete));
+      }
+    };
+
+    return toRemoveTodo();
+  }, [todos]);
+
+  const onUpdate = useCallback((todoID: number, data: Partial<Todo>) => {
+    setProcessingIds(prev => [...prev, todoID]);
+
+    const toUpdateTodo = async () => {
+      try {
+        await updateTodo(todoID, data);
+        getTodosFromServer();
+      } catch {
+        setCurrentError(Error.UPDATE);
+      }
+    };
+
+    toUpdateTodo();
+
+    setProcessingIds(processingIds.filter(id => id !== todoID));
+  }, [todos]);
+
+  const handleSubmit = useCallback((e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (!user) {
+      return;
+    }
+
+    if (!tempTodo.title.trim()) {
+      setCurrentError(Error.TITLE);
+
+      return;
+    }
+
+    onAdd({ ...tempTodo, userId: user.id });
+  }, [tempTodo]);
+
+  const onClear = useCallback(() => {
+    visibleTodos
+      .forEach(todo => (todo.completed ? onDelete(todo.id) : todo));
+  }, [visibleTodos]);
+
+  const toggleAll = useCallback(() => {
+    if (!isArrow) {
+      todos.map((todo: Todo) => (!todo.completed
+        ? onUpdate(todo.id, { completed: true })
+        : todo));
+
+      setIsArrow(true);
+    }
+
+    if (isArrow) {
+      todos.map((todo: Todo) => (todo.completed
+        ? onUpdate(todo.id, { completed: false })
+        : todo));
+
+      setIsArrow(false);
+    }
```

Comment:
use Promise.All for this
```suggestion
      Promise.All(todos.map((todo: Todo) => (!todo.completed
        ? onUpdate(todo.id, { completed: !isArrow })
        : todo));

      setIsArrow(!isArrow);
```